src/components/implementation/torrent/ramfs/ramfs.c

```diff
@@ -23,6 +24,43 @@ struct fsobj root;
 #define UNLOCK() if (lock_release(&fs_lock)) BUG();
 
 #define MIN_DATA_SZ 256
+#define DEBUG 0
+
+struct file_data {
+	cbuf_t cbuf_id;
+	char *chunk_start;
+	int cbuf_len;
+	int start;
+	int len;
+	void* next;
+};
+
+int file_data_init(struct file_data **fd, 
+		cbuf_t cbuf_id, 
+		char* chunk_start, 
+		int cbuf_length, 
+		int start, 
+		int len, 
+		void* next)
+{
+	assert(*fd);
+	(*fd)->cbuf_id = cbuf_id;
+	(*fd)->chunk_start = chunk_start;
+	(*fd)->cbuf_len = cbuf_length;
+	(*fd)->start = start;
+	(*fd)->len = len;
+	(*fd)->next = next;
+
+	return 0;
+}
+
+int file_data_alloc(struct file_data **fd, cbuf_t cbuf_id, char* chunk_start, int cbuf_length, int start, int len, void* next)
```

Comment:
you should just return a `struct file_data *`.  That makes all of this less awkward.